presto-main/src/main/java/com/facebook/presto/operator/SortedPositionLinks.java

```diff
@@ -192,53 +193,54 @@ public int next(int position, int probePosition, Page allProbeChannelsPage)
     @Override
     public int start(int startingPosition, int probePosition, Page allProbeChannelsPage)
     {
+        int[] links = sortedPositionLinks[startingPosition];
+        if (links == null) {
+            return -1;
```

Comment:
Isn't `links==null` the case of no collisions?

In original code, we had the following here:
```
if (applyLessThanFunction(startingPosition, probePosition, allProbeChannelsPage)) {
            return startingPosition;
}
```

and this seemingly could return `startingPosition` even if `sortedPositionLinks[startingPosition]==null`